Docs: make retryable errors configurable

infraredgirl opened the pull request:
Related issue: #1040.

brikis98 commented on docs/_docs/02_features/auto-retry.md:
```diff
@@ -24,10 +24,23 @@ Terraform can fail with transient errors which can be addressed by simply retryi
-Known errors that `auto-retry` will rerun, are maintained in the `TerragruntOptions.RetryableErrors` array. Future upgrades to `terragrunt` may include the ability to configure `auto-retry` by specifying additional error strings and configuring max retries and retry intervals the `terragrunt` config (PRs welcome\!).
+A list of default known errors that `auto-retry` will rerun is maintained in the `TerragruntOptions.RetryableErrors` array. These defaults can be overridden by specifying a list of retryable errors in the `terragrunt` config using the following format:
+
+    retryable_errors = {
```
NIT: we typically use [fenced code blocks](https://help.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks) for code. One advantage is that you can set the language to use for highlighting—in this case, `hcl`.

infraredgirl replied:
Sounds good, I also prefer fenced code blocks. Was just trying to mirror the existing style in this file (line 21). Will fix.

brikis98 commented on docs/_docs/02_features/auto-retry.md:
```diff
@@ -24,10 +24,23 @@ Terraform can fail with transient errors which can be addressed by simply retryi
 `auto-retry` will try a maximum of three times to re-run the command, at which point it will deem the error as not transient, and accept the terraform failure. Retries will occur when the error is encountered, pausing for 5 seconds between retries.
 
-Known errors that `auto-retry` will rerun, are maintained in the `TerragruntOptions.RetryableErrors` array. Future upgrades to `terragrunt` may include the ability to configure `auto-retry` by specifying additional error strings and configuring max retries and retry intervals the `terragrunt` config (PRs welcome\!).
+A list of default known errors that `auto-retry` will rerun is maintained in the `TerragruntOptions.RetryableErrors` array. These defaults can be overridden by specifying a list of retryable errors in the `terragrunt` config using the following format:
```
NIT: Most users don't look at the Terragrunt source code, so they won't know what `TerragruntOptions.RetryableErrors` is. Here's a different phrasing:

```suggestion
Terragrunt has a small list of default known errors built-in. You can override these defaults with your own custom retryable errors in your `terragrunt.hcl` configuration:
```

infraredgirl replied:
Makes total sense, I'll update.

infraredgirl commented:
Thanks for the review! I updated the PR to include review suggestions.

yorinasub17 commented:
We also need to include this new attribute in https://terragrunt.gruntwork.io/docs/reference/config-blocks-and-attributes/

infraredgirl commented:
Added a reference to the new attribute to the "Configuration Blocks and Attributes" doc page and squashed everything into a single commit. This is now ready for review.

brikis98 reviewed (approved):
Doc changes LGTM!

I forget, was the code for this in a separate PR?

infraredgirl commented:
> Doc changes LGTM!
> 
> I forget, was the code for this in a separate PR?

Yes, it's in the PR #1232. I can merged them together if that is preferred?

brikis98 commented:
> > Doc changes LGTM!
> > I forget, was the code for this in a separate PR?
> 
> Yes, it's in the PR #1232. I can merged them together if that is preferred?

Yea, you can either (a) merge them together and open a single PR, or (b) create a short-lived feature branch and open these PRs against it, rather than `master`, and then merge the feature branch to `master` at the end.

infraredgirl commented:
Makes sense! I'll add this doc change to the PR #1232 and close this PR then.

brikis98 commented:
Roger, thx!

infraredgirl commented:
Doc changes now live in #1232. Closing this.
